torch/csrc/jit/tensorexpr/ir_simplifier.cpp

```diff
@@ -988,13 +988,162 @@ const Expr* PolynomialTransformer::mutate(const Div* v) {
     return getImmediateByType(v->dtype(), 1);
   }
 
+  auto* lhsTerm = dynamic_cast<const Term*>(lhs_new);
+  auto* rhsTerm = dynamic_cast<const Term*>(rhs_new);
+
+  if (const Polynomial* lhsPoly = dynamic_cast<const Polynomial*>(lhs_new)) {
+    if (auto* term = factorizePolynomial(lhsPoly)) {
+      lhsTerm = term;
+    }
+  }
+  if (const Polynomial* rhsPoly = dynamic_cast<const Polynomial*>(rhs_new)) {
+    if (auto* term = factorizePolynomial(rhsPoly)) {
+      rhsTerm = term;
+    }
+  }
+
+  if (lhsTerm && !rhsTerm) {
+    rhsTerm =
+        new Term(hasher_, getImmediateByType(rhs_new->dtype(), 1), {rhs_new});
+  } else if (rhsTerm && !lhsTerm) {
+    lhsTerm =
+        new Term(hasher_, getImmediateByType(lhs_new->dtype(), 1), {lhs_new});
+  }
+
+  if (lhsTerm && rhsTerm) {
+    auto lVars = lhsTerm->variables();
+    auto rVars = rhsTerm->variables();
+    bool didAnything = false;
+
+    // we want to find the difference between these vectors, which are sorted by
+    // hash.
+    for (auto lIt = lVars.begin(); lIt != lVars.end() && !rVars.empty();) {
+      auto lHash = hasher_.hash(*lIt);
+      bool incrementedL = false;
+      for (auto rIt = rVars.begin(); rIt != rVars.end();) {
+        auto rHash = hasher_.hash(*rIt);
+        if (lHash == rHash) {
+          lIt = lVars.erase(lIt);
```

Comment:
I'd add a note here that it's valid to continue iteration from the iterator `erase` returns, I hesitated for a moment when reading the code.

Reply:
Sure, happy to add a comment. This is a pattern I use often when deleting from a container with a nontrivial condition, so it's likely in other places in the codebase.